pkg/services/hivelocity/device/device.go

```diff
@@ -168,13 +171,26 @@ func GetFirstFreeDevice(ctx context.Context, hvclient hvclient.Client, machineTy
 		return devices[i].DeviceId < devices[j].DeviceId
 	})
 
-	device := findAvailableDeviceFromList(devices, machineType, hvCluster.Name)
-	return device, nil
+	hvDevices := findAvailableDeviceFromList(devices, hvMachineSpec.DeviceSelector, hvCluster.Name)
```

Comment:
can you also check here again what is a rebase problem and what is not? 
You rename currently "device" into "hvDevices" which seems a bit useless to me, and even wrong in the plural form. 
Then the question is whether you added these conditions here in this PR or whether they exist already.